### PR TITLE
mfem, pflotran, alquimia: remove old versions with xsdk string (in version)

### PR DESCRIPTION
**Pull request by balay**
These old pkg versions were used in older xsdk releases - that are already removed from spack.

### CODE REVIEW

**Review by smolins — APPROVED** [review]

**Review by tldahlgren — COMMENTED** [review]
LGTM though I do wonder if you want to flag them as deprecated first.

There's guidance at https://spack.readthedocs.io/en/latest/packaging_guide.html#deprecating-old-versions.

**Comment by tldahlgren** [comment]
I would also like to give at least one `mfem` maintainer a chance to comment.

**Comment by tldahlgren** [comment]
FYI. I also grep'd dependencies of built-in packages to double-check that all occurrences of said versions are removed.

**Comment by balay** [comment]
> LGTM though I do wonder if you want to flag them as deprecated first.

This is more of a cleanup - the corresponding xsdk versions are already deprecated and removed. [and these pkgs should also have been deprecated and removed in sync with these xsdk versions - but this didn't get done at that time]. So  I think   removing them now is more appropriate.

**Review by ghammond86 — APPROVED** [review]

**Comment by balay** [comment]
ping @v-dobrev @tzanio - it would be great if you can review/approve.

**Review by tldahlgren — APPROVED** [review]

**Comment by tldahlgren** [comment]
> ping @v-dobrev @tzanio - it would be great if you can review/approve.

Apologies.  I didn't check that auto merge was enabled before I approved the PR.

**Pull request merged**
